### PR TITLE
Change api of ProductName to use render props

### PR DESCRIPTION
**Pull request by Klynger**
#### What is the purpose of this pull request?
Update the API of `ProductName` to be using render props

#### What problem is this solving?
Make `ProductName` more customizable

#### How should this be manually tested?
[Workspace](https://productnamerenderprops--storecomponents.myvtex.com/)

#### Screenshots or example usage
captura de tela de 2018-11-01 10-36-34

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ProductName/index.js [hunk 1]
```diff
@@ -25,19 +24,31 @@ class ProductName extends Component {
...
+    renderProductReference: PropTypes.func
   }
 
   static defaultProps = {
     large: false,
     showBrandName: false,
     showProductReference: false,
     showSku: false,
+    loaderClasses: '',
+    rootClasses: ''
```
Comment (luciannojunior) [inline react/components/ProductName/index.js]: no default for the render methods?

Comment (luciannojunior) [reply react/components/ProductName/index.js]: I mean, for a `ProductName` component I would expect it to at least render a span with the name I passed... I don't know if you already discussed it with @brunojdo

Comment (Klynger) [reply react/components/ProductName/index.js]: You mean like
```javascript
// code...
static defaultProps = {
  //...
  renderName: ({ name }) => <span>{name}</span>
}
```
?

File: react/components/ProductName/index.js [hunk 2]
```diff
@@ -67,44 +78,31 @@ class ProductName extends Component {
...
-        {showProductReference && productReference && (
-          <div className="vtex-product-name__product-reference pt3 f7 ttu">
-            {`REF: ${productReference}`}
-          </div>
-        )}
+      <div className={classNames('vtex-product-name', rootClasses)}>
+        {renderName && renderName({ large, name })}
+        {showSku && renderSku && renderSku({ skuName, large })}
+        {showProductReference && renderProductReference
+          && renderProductReference({ productReference, large })}
```
Comment (klzns) [inline react/components/ProductName/index.js]: O que acha de ter apenas uma render prop:

```js
render() {
  return this.props.children({
    large,
    name,
    skuName: showSku ? skuName: null,
    productReference: showProductReference ? productReference : null,
  })
}
```

Dessa forma a ordem que vai ser renderizada esses elementos é definida pelo componente cliente.

File: react/components/ProductName/README.md [hunk 3]
```diff
@@ -9,22 +9,40 @@ import ProductName from 'vtex.store-components/ProductName'
...
-    name={product.productName}
-    skuName={selectedItem.name}
-    brandName={product.brand}
+  name={product.productName}
+  skuName={selectedItem.name}
+  brandName={product.brand}
+  renderName={({ name, large }) => {
+      let classes = 'vtex-product-name__brand c-on-base'
+      if (large) {
+        classes = classNames(classes, 'some-css-class-for-large-typograhy')
```
[the remark this answers is not in the document]
Comment (Klynger) [reply react/components/ProductName/README.md]: Acho que vale a pena usar o classNames pq dá pra usar a prop `large` em conjunto

**Pull request closed without merging**